src/yunohost/data_migrations/0009_dh_params.py

```diff
@@ -0,0 +1,36 @@
+import re
+import os
+
+from moulinette.utils.log import getActionLogger
+from moulinette.utils.filesystem import chown
+
+from yunohost.tools import Migration
+from yunohost.service import service_regen_conf
+
+command = "nice -n 19 openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048 -dsaparam 2> /var/log/yunohost/dhparam_generation.log && chown root:ssl-cert /etc/ssl/private/dh2048.pem && rm /etc/cron.hourly/yunohost-generate-dh-params\n"
+dhparams_file = "/etc/ssl/private/dh2048.pem"
+
+class MyMigration(Migration):
+    "This migration will add dh_params line and generate it in installed instance"
+
+    def migrate(self):
+
+        try:
+            file_open = open(dhparams_file)
+            service_regen_conf(['nginx'], force=True)
```

Comment:
I really don't think you want to do this :wink: It's only relevant to regen nginx *after* the DH generation is done, so it should happen in the cron job itself.

Also you should absolutely not use `--force` as it's gonna override manual changes, which may break some critical stuff for people who customized stuff.

Reply:
Normally, here it's the case where dh_params is already  generated. So I would say that it's useful here, am I wrong ?

Reply:
I've added it in command. Here there is not cron... do it's normal. Or, perhaps I didn't understand what you wanted  ?